src/main/java/ca/pjer/logback/AwsLogsAppender.java

```diff
@@ -112,24 +115,39 @@ public synchronized void stop() {
             layout.stop();
         }
     }
+    
+    private final List<InputLogEvent> backLog = new LinkedList<InputLogEvent>();
 
     @Override
     protected void append(ILoggingEvent event) {
+        final InputLogEvent inputLogEvent = new InputLogEvent()
+                .withTimestamp(event.getTimeStamp())
+                .withMessage(layout.doLayout(event));
         try {
-            PutLogEventsRequest request = new PutLogEventsRequest()
+            
+            final PutLogEventsRequest request = new PutLogEventsRequest()
                     .withLogGroupName(logGroupName)
                     .withLogStreamName(logStreamName)
-                    .withSequenceToken(sequenceToken)
-                    .withLogEvents(new InputLogEvent()
-                            .withTimestamp(event.getTimeStamp())
-                            .withMessage(layout.doLayout(event)));
-            PutLogEventsResult result = awsLogs.putLogEvents(request);
+                    .withSequenceToken(sequenceToken);
+            if (backLog.size() > 0) {
+                request.withLogEvents(backLog);
+            }
+            request.withLogEvents(inputLogEvent);
+            final PutLogEventsResult result = awsLogs.putLogEvents(request);
             sequenceToken = result.getNextSequenceToken();
+            backLog.clear();
         } catch (DataAlreadyAcceptedException e) {
             sequenceToken = e.getExpectedSequenceToken();
         } catch (InvalidSequenceTokenException e) {
             sequenceToken = e.getExpectedSequenceToken();
             append(event);
+        } catch (AmazonServiceException e) {
+            backLog.add(inputLogEvent);
```

Comment:
The Exception that I receive here is just the AmazonServiceException, the only way to check the Throthling is by checking the error Code String.

You are correct about other problems that can fill the backLog, maybe we can add a limit to the backLog, I'm using this logger in one web application and getting many times this exception, probably you can test in your environment creating a loop generating log messages to test.

The best way probably will be to add configuration for sending in batches. The problem That I see is some messages will be waiting until the buffer fill, and for applications with low logging some messages will not be sent to cloudwatch logs.

I did not see anything in logback that permit to create and send a batch of messages.